homeassistant/components/syncthing/sensor.py

```diff
@@ -266,3 +283,66 @@ def _filter_state(self, state):
         state["label"] = self._folder_label
 
         return state
+
+
+class SpeedSensor(CoordinatorEntity):
+    """A syncthing speed sensor."""
+
+    NAMES = {"download": "inBytesTotal", "upload": "outBytesTotal"}
+
+    def __init__(self, coordinator, url, server_id, name, version):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._name = name
+        self._last_total_value = self._current_value()
+        self._last_updated = monotonic()
+        self._state = 0
+
+        short_server_id = server_id.split("-")[0]
+        self._attr_name = f"{short_server_id} {self._name}"
+        self._attr_unique_id = f"{short_server_id}-{self._name}"
+        self._attr_icon = "mdi:speedometer"
+        self._attr_unit_of_measurement = DATA_RATE_MEGABYTES_PER_SECOND
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, server_id)},
+            "name": f"Syncthing ({url})",
+            "manufacturer": "Syncthing Team",
+            "sw_version": version,
+            "entry_type": "service",
+        }
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if not self.available:
+            return
+        return self._state
+
+    @property
+    def device_info(self):
+        """Return device information."""
+        return
```

Comment:
Device info is set as `_attr_device_info` in the `__init__`.
However, this property overrides it?